packages/labs/motion/src/animate.ts

```diff
@@ -420,6 +421,7 @@ export class Animate extends AsyncDirective {
         // TODO(sorvell): these nudges could conflict with existing styling
         // or animation but setting left/top should be rare, especially via
         // animation.
+        this._fromValues ??= disconnectedFromValues;
```

Comment:
I think assigning back to `this` is a little hard to reason about. If directive instance state might change in bad ways while awaiting an animation frame, then I think that the critical state should be saved off before the `await animationFrame()` and reference to `this` be avoided after it, to avoid the "shearing".

Even if you don't do this for every field yet, I think it'd be good to structure it this way for `_fromValues`:

```ts
    this.prepare();
    const fromValues = this._fromValues;
    await animationFrame();
    // ...
    const left = fromValues['left'] ...
```